### ISSUE TITLE
update karaf to 4.0.3

### ISSUE THREAD

**Issue by ruebot** [Islandora-CLAW/CLAW#57]

**Comment by acoburn** [comment]
Just as a note, the latest Karaf release is now 4.0.2

**Comment by ruebot** [comment]
I'll update the ticket :smile:

**Comment by daniel-dgi** [comment]
now it's 4.03 :D

**Comment by daniel-dgi** [comment]
Heads up, this will break the basic image service because karaf 4's jetty component has issues with multipart/form-data.  So we have two options: wait, or figure out a better services API :P 

I say we should have a big long talk about our services API, and if we can do it differently.  We could expose transactionality in Fedora and just have multiple calls within a transaction as opposed to accepting multipart/form-data.  That is, open a transaction, ingest a basic image, add its OBJ, close transaction.

It may be best we take an approach like that.  There's also the option of providing a single file for ingest/update, like a zipped bag or something.

**Comment by DiegoPino** [comment]
@daniel-dgi, true. Let's figure it out (or start) during the Sprint. I'm still trying to understand what is missing in jetty9 to make multipart happen, if not a bug.

**Comment by ruebot** [comment]
Sounds like a good topic for our Wednesday call! I'll add it to the agenda :smile:

https://github.com/Islandora-Labs/islandora/wiki/November-4,-2015

**Comment by daniel-dgi** [comment]
Ok, looks like if I throw in a JAXRS annotated interface into the collection and image services, we can hijack the route using camel-cxfrs.  This will use karaf's internal jetty server (the one hawt is running on) instead of trying to have the camel context fire up its own.  Will also give us more solid multipart support.  Plus swagger :D

The intended use of something like this is to provide middleware services around cxf web services that are already defined in service beans.  Ideally, we'd have more done in code in service beans or processors for a bit better code re-use.  But that's out of scope for this issue.  For now, this approach will get us what we need to upgrade to newer to unblock all the other issues that are waiting for this.

**Comment by ruebot** [comment]
EXCELLENT.

**Comment by ruebot** [comment]
Resolved with: fa60702349e809f82d27b084f3fc8c82fe570109
